### ISSUE TITLE
UNRAID build: change webui password

### ISSUE THREAD

**Issue by faustusdotbe** [datamachines/cuda_tensorflow_opencv#29]
Hi! Thanks for this.

I am wondering if you could help me find out how to change the password of the webui?

I've tried these things: 
- adding a new variable within the Unraid template
- manually changing the value in `/home/jupyter/.jupyter/jupyter_notebook_config.json`

.. to no avail. 

Any idea? Thanks

**Comment by mmartial** [comment]
If you change the password in a running instance, it will not work until you restart the container. 
I tested and changed it to `password` by:
1) Starting a "New" "Terminal" in the currently running NoteBook
2) Using `jupyter notebook password` and entering the new password.
3) In the Unraid Docker UI, `Stop` the container, then `Start`ing it again
4) When accessing the `WebUI` the new `password` will let you in.

Manually changing the value in `/home/jupyter/.jupyter/jupyter_notebook_config.json` should work as well (it is what the `Dockerfile-unraid` does: `COPY` an entry with a default password); you just need to restart the Container.

tl;dr
The original plan was to have the `/home/jupyter` accessible to users in the Unraid template, but that prevented the addition of a password and would have needed users to obtain the startup token from the logs.

The `Dockerfile` for Unraid is slightly different from the `-user` one to support its default `uid` and `gid` natively and setup a user password.

IF you are curious, the file itself is at 
https://github.com/datamachines/cuda_tensorflow_opencv/blob/master/Jupyter_build/Dockerfile-unraid
and the setup is documented at https://github.com/datamachines/cuda_tensorflow_opencv#BuildofanUnraidspecificJupyterNotebookcontainer

**Comment by faustusdotbe** [comment]
Thanks for this! Unfortunately I get a `permissionError` when I try to do the terminal way in the notebook:

![image](https://user-images.githubusercontent.com/5579081/161446774-56eb7968-bac2-4b3f-9c52-3c1feb73b02e.png)

It's a bit late in the evening here so I'll try debugging tomorrow. 

Thanks for linking the Dockerfile! I'm not super familiar with Docker yet but eager to learn, so that is quite useful.

**Comment by mmartial** [comment]
I did get that too, I thought it was some other manual changes I did.
`sudo chown jupyter.users /home/jupyter/.jupyter/jupyter_notebook_config.json` will bypass this issue.
I will check why the file is created as `root` (it is created by the `jupyter` user after all)

**Comment by mmartial** [comment]
The `COPY` is ran as `root`, so that exact `chown` will need to be added in the `Dockerfile-unraid` for future releases. Thanks for finding this.

**Comment by mmartial** [comment]
Added `RUN sudo chown jupyter.users /home/jupyter/.jupyter/jupyter_notebook_config.json` right after the `COPY`, started the container and I can confirm the file has the right user at this point
```
jupyter@d0fdeda670a3:~$ ls -al .jupyter/
total 68
drwx------ 1 jupyter users  4096 Apr  3 21:09 .
drwxr-xr-x 1 jupyter users  4096 Apr  3 21:09 ..
-rw-rw-r-- 1 jupyter users   161 Apr  2 18:39 jupyter_notebook_config.json
-rw-r--r-- 1 jupyter users 54172 Apr  3 21:09 jupyter_notebook_config.py
```

Making a commit, uploading a new version to DockerHub and then closing the ticket.

Thanks for the report.

**Comment by faustusdotbe** [comment]
This fixed it, thanks @mmartial !
